lib/index.ts

```diff
@@ -378,17 +409,17 @@ export class IgnoreType implements IGetToken<void> {
   }
 
   // ToDo: don't read, but skip data
-  public get(buf: Buffer, off: number) {
+  public get(array: Uint8Array, off: number) {
   }
 }
 
-export class BufferType implements IGetToken<Buffer> {
+export class Uint8ArrayType implements IGetToken<Uint8Array> {
 
   public constructor(public len: number) {
   }
 
-  public get(buf: Buffer, off: number): Buffer {
-    return buf.slice(off, off + this.len);
+  public get(array: Uint8Array, off: number): Uint8Array {
+    return array.slice(off, off + this.len);
```

Comment:
While this works just fine, The buffer slice differ from TypedArray slice

> This method is not compatible with the Uint8Array.prototype.slice(), which is a superclass of Buffer. To copy the slice, use Uint8Array.prototype.slice().
> https://nodejs.org/dist/latest-v15.x/docs/api/buffer.html#buffer_buf_slice_start_end

- Buffer slice will use the same underlying ArrayBuffer with a new offset/length and match kind of what uint8.subarray use instead
- TypedArray slice will actually copy the data

So to match the same behavior as `buffer.slice()` you should actually call the [subarray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/subarray) method

Think it was stupid of NodeJS to override that slice method and not be compatible in that regards, ppl should just have used subarray instead if they wanted to do so.

Without reading node's docs i would actually have expected that slice would give me a new copy - not using the same memory with another offset...